modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLPreviewLimitManagedServiceFactory.java

```diff
@@ -0,0 +1,514 @@
+		}
+
+		return systemValue;
+	}
+
+	private long _getMinimumMaxFileSize(long companyValue, long systemValue) {
+		return _getMinimum(
+			companyValue, _UNLIMITED_MAX_FILE_SIZE, systemValue,
+			_UNLIMITED_MAX_FILE_SIZE);
+	}
+
+	private long _getMinimumMaxFileSize(
+		long companyValue, long groupValue, long systemValue) {
+
+		return _getMinimum(
+			companyValue, groupValue, systemValue, _UNLIMITED_MAX_FILE_SIZE);
+	}
+
+	private int _getMinimumMaxNumberOfPages(int companyValue, int systemValue) {
+		return (int)_getMinimum(
+			companyValue, _UNLIMITED_MAX_NUMBER_OF_PAGES, systemValue,
+			_UNLIMITED_MAX_NUMBER_OF_PAGES);
+	}
+
+	private int _getMinimumMaxNumberOfPages(
+		int companyValue, int groupValue, int systemValue) {
+
+		return (int)_getMinimum(
+			companyValue, groupValue, systemValue,
+			_UNLIMITED_MAX_NUMBER_OF_PAGES);
+	}
+
+	private Configuration _getScopedConfiguration(
+			ExtendedObjectClassDefinition.Scope scope, long scopePK)
+		throws Exception {
+
+		Configuration[] configurations = _configurationAdmin.listConfigurations(
+			String.format(
+				"(&(service.factoryPid=%s)(%s=%d))",
+				DLPreviewLimitConfiguration.class.getName() + ".scoped",
+				scope.getPropertyKey(), scopePK));
+
+		if (configurations == null) {
+			return null;
+		}
+
+		return configurations[0];
+	}
+
+	private boolean _isLimitExceeded(
+		long limit, long value, long unlimitedValue) {
+
+		if ((limit != unlimitedValue) &&
+			((value == unlimitedValue) || (limit < value))) {
+
+			return true;
+		}
+
+		return false;
+	}
+
+	private void _unmapPid(String pid) {
+		if (_companyIds.containsKey(pid)) {
+			long companyId = _companyIds.remove(pid);
+
+			_companyConfigurationBeans.remove(companyId);
+		}
+
+		if (_groupIds.containsKey(pid)) {
+			long groupId = _groupIds.remove(pid);
+
+			_groupConfigurationBeans.remove(groupId);
+		}
+	}
+
+	private void _updateCompanyConfiguration(
+		long companyId, String pid, Dictionary<String, ?> dictionary) {
+
+		_companyConfigurationBeans.put(
+			companyId,
+			ConfigurableUtil.createConfigurable(
+				DLPreviewLimitConfiguration.class, dictionary));
+		_companyIds.put(pid, companyId);
+	}
+
+	private void _updateCompanyDLPreviewLimitConfiguration(
+			long companyId, long maxFileSize, int maxNumberOfPages)
+		throws Exception {
+
+		_updateScopedConfiguration(
+			maxFileSize, maxNumberOfPages,
+			ExtendedObjectClassDefinition.Scope.COMPANY, companyId);
+	}
+
+	private void _updateGroupConfiguration(
+		long groupId, String pid, Dictionary<String, ?> dictionary) {
+
+		_groupConfigurationBeans.put(
+			groupId,
+			ConfigurableUtil.createConfigurable(
+				DLPreviewLimitConfiguration.class, dictionary));
+		_groupIds.put(pid, groupId);
+	}
+
+	private void _updateGroupDLPreviewLimitConfiguration(
+			long groupId, long maxFileSize, int maxNumberOfPages)
+		throws Exception {
+
+		_updateScopedConfiguration(
+			maxFileSize, maxNumberOfPages,
+			ExtendedObjectClassDefinition.Scope.GROUP, groupId);
+	}
```

Comment:
Don't you think this method (and the one for company) is a bit redundant? It is doing nothing of value; can you inline it?

Reply:
Done.